### ISSUE TITLE
Loading entire dataset into memory

### ISSUE THREAD

**Issue by maxsolomonhenry** [dgsmith1988/ECSE-552-Final-Project#2]
@dgsmith1988 

This is fast, no doubt, but will crash if our dataset gets larger than 5Gb.  David, the TA, suggested (in addition to trying to load everything into memory) that we play with the num_workers in the dataloder setting as an alternative (see forwarded e-mail).

We may not have to worry about this for a while, but just raising it now so it stays on our radar.

https://github.com/dgsmith1988/ECSE-552-Final-Project/blob/31867faac984ce5ce236be5702af252f4bbf287e/Code/feature_extraction.py#L123

**Comment by dgsmith1988** [comment]
Yeah, I forgot about the number of workers aspect and I'm sure there are plenty of optimizations which could be made. My main goal with this was to get something up and running so we could make more progress in terms of the experimentation and data collection so I didn't touch a lot of the previous code if it worked well enough. I'm not sure I fully understand where the 5GB limitation is coming from but I could see that if there are limitations to the built-in data types for Python.

There are plenty of opportunities for optimization so go nuts now that we have a baseline set of behavior we can compare against. Librosa could be replaced with Pytorch to take advantage of the GPU for the spectrogram transforms which would help for scaling to larger amounts of data as it's only ~7 minutes to pre-process the data at this point.

To my knowledge the data is all loaded into memory, however if that is not the case feel free to point it out.

I'd suggest that we explore the model more within the context of the data that is currently available before expanding at this point. If things to get a point where not all the data fits into memory, then the swapping should be a lot faster as all the data is localized to the Colab Instance at first so the data would be going between the RAM and the HDD (and no network calls would be necessary).
